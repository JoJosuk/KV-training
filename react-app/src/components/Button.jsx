import React from "react";
const Button = ({ buttonContent, loggedin }) => {
  return (
    <button
      type="submit"
      data-testid="button-component"
      onClick={(e) => {
        e.preventDefault();
        loggedin();
      }}
    >
      {buttonContent}
    </button>
  );
};
export default Button;
