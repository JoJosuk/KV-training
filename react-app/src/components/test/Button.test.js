import React from "react";
import { fireEvent, render } from "@testing-library/react";
import Button from "../Button";
describe("check if button works properly", () => {
  test("Check if Button rendered correctly ", () => {
    const { getByTestId } = render(<Button />);
    const element = getByTestId("button-component");

    expect(element).toBeTruthy();
  });

  test("check if text displayed properly", () => {
    const text = "Click";
    const { getByText } = render(<Button buttonContent={text} />);
    getByText(text);
  });

  test("check if onClick is triggered", () => {
    const onClick = jest.fn();
    const { getByTestId } = render(<Button loggedin={onClick} />);
    const element = getByTestId("button-component");

    fireEvent.click(element);

    expect(onClick).toHaveBeenCalledTimes(1);
  });

  test("check if the snapshot is matched properly", () => {
    const onClick = jest.fn();
    const text = "Click";

    const { asFragment } = render(
      <Button buttonContent={text} loggedin={onClick} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
