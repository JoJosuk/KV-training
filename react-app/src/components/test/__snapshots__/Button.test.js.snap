// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if button works properly check if the snapshot is matched properly 1`] = `
<DocumentFragment>
  <button
    data-testid="button-component"
    type="submit"
  >
    Click
  </button>
</DocumentFragment>
`;
